apps/intellij/src/main/kotlin/dev/nx/console/angular/NxAngularConfigService.kt

```diff
@@ -0,0 +1,111 @@
+package dev.nx.console.angular
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.EDT
+import com.intellij.openapi.components.*
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.RootsChangeRescanningInfo
+import com.intellij.openapi.roots.ex.ProjectRootManagerEx
+import com.intellij.openapi.util.EmptyRunnable
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.VirtualFileManager
+import dev.nx.console.nxls.NxWorkspaceRefreshListener
+import dev.nx.console.nxls.NxlsService
+import dev.nx.console.nxls.NxlsService.Companion.NX_WORKSPACE_REFRESH_TOPIC
+import dev.nx.console.utils.writeAction
+import java.nio.file.Path
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.angular2.cli.config.AngularConfig
+import org.angular2.cli.config.AngularConfigProvider
+
+@Service(Service.Level.PROJECT)
+@State(name = "NxAngularConfigService", storages = [Storage("nx-angular-config.xml")])
+class NxAngularConfigService(private val project: Project, private val cs: CoroutineScope) :
```

Comment:
Thanks to compatibility I can get the `couroutineScope` from the platform. It has correct lifecycle and gets cancelled when project or plugin are unloaded. I think you should convert other services to use injected scope instead of the ad hoc ones.